README.md

```diff
@@ -93,6 +93,37 @@ proxy.createProxy({
 });
 ```
 
+
+### Connection Information
+
+#### IP, Port of Client, Server and Self
+
+```js
+proxy.createProxy({
+  forwardPort: 9999,
+  interceptor: {
+    client(result, encoding, connection) {
+      console.info('Connection from ', connection.client.ip, ':', connection.client.port, 'to', connection.self.ip, ':', connection.self.port, 'from', connection.server.ip, ':', connection.server.port);
+      return result
+    }
+  },
+});
+```
+
+#### Data size
+
+```js
+proxy.createProxy({
+  forwardPort: 9999,
+  interceptor: {
+    server(result, encoding, connection) {
+      console.log('Connection Size:' + connection.size);
```

Comment:
Where is the `size` ?

Reply:
Hello, sorry I've been away for a little while but have a look at this

```
function genThrough(interceptor, connection) {

  return through.obj(function(chunk, enc, done) {

    let context = {
      client: {
        ip: connection.client.address().address,
        port: connection.client.address().port,
      },
      server : {
        ip: connection.server.address().address,
        port: connection.server.address().port,
      },
      self: {
        ip: connection.forwardHost,
        port: connection.forwardPort,
      },
      size : chunk.length,
    };

    const result = interceptor(chunk, enc, context);

    const handle = data => {

      if (data && !Buffer.isBuffer(data)) {
        data = Buffer.from(data);
      }

      done(null, data || chunk);
    };

    if (result) {
      if (typeof result.then === 'function') {
        result.then(handle).catch(e => {
          console.error(e);
          handle();
        });
      } else {
        handle(result);
      }
    } else {
      handle();
    }

  });

}
```

Reply:
Any updates @whxaxes or would you like anything changed/added ?

Reply:
No, Sorry! I forgot this PR for a long time, I will merged this. Thx